dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java

```diff
@@ -1325,4 +1341,113 @@ private String getAuthorizationID(String epersonUuid, String featureName, String
+                                .param("feature", loginOnBehalfOf.getName()))
+                    .andExpect(status().isOk())
+                    .andExpect(jsonPath("$._embedded.authorizations", Matchers.hasItem(
+                        AuthorizationMatcher.matchAuthorization(loginOnBehalfOfAuthorization))));
+    }
+
+    @Test
+    public void loginOnBehalfNonSiteObjectOfTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName("Parent Community")
+                                          .build();
+        context.restoreAuthSystemState();
+
+        CommunityRest communityRest = communityConverter.convert(parentCommunity, Projection.DEFAULT);
+        String communityUri = utils.linkToSingleResource(communityRest, "self").getHref();
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        configurationService.setProperty("org.dspace.app.rest.authorization.AlwaysThrowExceptionFeature.turnoff", true);
+        configurationService.setProperty("webui.user.assumelogin", true);
+
+        Authorization loginOnBehalfOfAuthorization = new Authorization(admin, loginOnBehalfOf, communityRest);
+        getClient(token).perform(get("/api/authz/authorizations/search/object")
+                                     .param("uri", communityUri)
+                                     .param("eperson", String.valueOf(admin.getID()))
+                                     .param("feature", loginOnBehalfOf.getName()))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.authorizations", Matchers.not(Matchers.hasItem(
+                            AuthorizationMatcher.matchAuthorization(loginOnBehalfOfAuthorization)))));
+    }
+
+    @Test
+    public void loginOnBehalfOfNonAdminUserNotFoundTest() throws Exception {
+        Site site = siteService.findSite(context);
+        SiteRest siteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(siteRest, "self").getHref();
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        configurationService.setProperty("org.dspace.app.rest.authorization.AlwaysThrowExceptionFeature.turnoff", true);
+        configurationService.setProperty("webui.user.assumelogin", true);
+
+        Authorization loginOnBehalfOfAuthorization = new Authorization(eperson, loginOnBehalfOf, siteRest);
+        getClient(token).perform(get("/api/authz/authorizations/search/object")
+                                     .param("uri", siteUri)
+                                     .param("eperson", String.valueOf(eperson.getID()))
+                                     .param("feature", loginOnBehalfOf.getName()))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.authorizations", Matchers.not(
```

Comment:
@abollini We can't test on an empty list here as some features are always returned by default. For example the "AlwaysTrueFeature". As we can't anticipate which features will be returned it would be best to check for an exclusion.